marimo/_runtime/runner/hooks_post_execution.py

```diff
@@ -0,0 +1,146 @@
+    should_send_output = (
+        run_result.output is not None or run_result.accumulated_output is None
+    )
+    if (
+        run_result.success()
+        or isinstance(run_result.exception, MarimoStopError)
+    ) and should_send_output:
+
+        def format_output() -> formatting.FormattedOutput:
+            formatted_output = formatting.try_format(run_result.output)
+            if formatted_output.traceback is not None:
+                write_traceback(formatted_output.traceback)
+            return formatted_output
+
+        if runner.execution_context is not None:
+            with runner.execution_context(cell.cell_id):
+                formatted_output = format_output()
+        else:
+            formatted_output = format_output()
+        CellOp.broadcast_output(
+            channel=CellChannel.OUTPUT,
+            mimetype=formatted_output.mimetype,
+            data=formatted_output.data,
+            cell_id=cell.cell_id,
+            status=cell.status,
+        )
+    elif isinstance(run_result.exception, MarimoInterrupt):
+        LOGGER.debug("Cell %s was interrupted", cell.cell_id)
+        # don't clear console because this cell was running and
+        # its console outputs are not stale
+        CellOp.broadcast_error(
+            data=[MarimoInterruptionError()],
+            clear_console=False,
+            cell_id=cell.cell_id,
+            status=cell.status,
+        )
+    elif run_result.exception is not None:
+        LOGGER.debug(
+            "Cell %s raised %s",
+            cell.cell_id,
+            type(run_result.exception).__name__,
+        )
+        # don't clear console because this cell was running and
+        # its console outputs are not stale
+        exception_type = type(run_result.exception).__name__
+        CellOp.broadcast_error(
+            data=[
+                MarimoExceptionRaisedError(
+                    msg="This cell raised an exception: %s%s"
+                    % (
+                        exception_type,
+                        (
+                            f"('{str(run_result.exception)}')"
+                            if str(run_result.exception)
+                            else ""
+                        ),
+                    ),
+                    exception_type=exception_type,
+                    raising_cell=None,
+                )
+            ],
+            clear_console=False,
+            cell_id=cell.cell_id,
+            status=cell.status,
+        )
+
+
+def _reset_matplotlib_context(
+    cell: CellImpl,
+    runner: cell_runner.Runner,
+    run_result: cell_runner.RunResult,
+) -> None:
+    del cell
+    del run_result
+    if get_global_context().mpl_installed:
+        # ensures that every cell gets a fresh axis.
+        exec("__marimo__._output.mpl.close_figures()", runner.glbls)
+
+
+POST_EXECUTION_HOOKS = [
+    _set_status_idle,
+    _broadcast_variables,
+    _send_output_to_frontend,
+    _reset_matplotlib_context,
```

Comment:
nit: the consistency of the top 3 hooks feel similar but off, `_set_status_idle`, `_broadcast_variables`, `_send_output_to_frontend`

 at least the last two could be `_broadcast_variables`, `_broadcast_outputs`